src/scripts/main.js

```diff
@@ -1,3 +1,38 @@
 'use strict';
 
-// write code here
+const appendRow = document.querySelector('.append-row');
+const removeRow = document.querySelector('.remove-row');
+const appendColumn = document.querySelector('.append-column');
+const removeColumn = document.querySelector('.remove-column');
+
+const table = document.querySelector('tbody');
+
+appendColumn.addEventListener('click', () => {
+  if ([...table.rows[0].cells].length < 10) {
+    [...table.rows].forEach((row) => {
+      const cloneColumn = document.createElement('td');
+
+      row.append(cloneColumn);
+    });
+  }
+});
+
+removeColumn.addEventListener('click', () => {
+  if ([...table.rows[0].cells].length > 2) {
+    [...table.rows].forEach((row) => {
+      row.deleteCell(table.rows[0].cells.length - 1);
+    });
+  }
+});
+
+appendRow.addEventListener('click', () => {
+  if ([...table.rows].length < 10) {
+    table.append(table.rows[0].cloneNode(true));
+  }
+});
+
+removeRow.addEventListener('click', () => {
+  if ([...table.rows].length > 2) {
+    table.deleteRow(table.rows[0]);
```

Comment:
The `deleteRow` method should take an index as its argument, not a row element. You should pass the index of the row you want to delete, such as `table.deleteRow(0)` to delete the first row.